NS_Install2.sh

```diff
@@ -84,31 +84,60 @@ sleep 10
 done
 EOF
 
-cs=`grep 'API_SECRET=' /etc/nsconfig | head -1 | cut -f2 -d'"'`
-
-echo "Current API secret is: $cs"
-echo
-echo "If you would like to change it please enter the new secret now or hit enter to leave the same"
+cs2=`grep 'API_SECRET=' /etc/nsconfig | cut -s -f2 -d'"'` # This is API_SECRET if double quotes are used in nsconfig.
+cs1=`grep 'API_SECRET=' /etc/nsconfig | cut -s -f2 -d\'` # This is API_SECRET if single quotes are used in nsconfig.
+cs="$cs2"
+if [ "$cs2" = "" ]
+then
+  cs="$cs1" # This is the current secret (API_SECRET) from nsconfig.
+fi
 
-for j in {1..1000}
```

Comment:
what happened to this part?

Reply:
We're now using a dialog for the API_SECRET.

I understood that was needed when using a text prompt for asking for the APPI_SECRET.

Did I misunderstand?

Reply:
What happens when someone presses a lot of chars before reaching this stage. Are they being removed?

Reply:
I don't have a maximum for the number of characters anywhere in any of the dialogs.

Do we need to specify a maximum?  
I can add that to the prompt asking user to enter an API_SECRET that has 12-24 characters for example.

Should I di that?

Reply:
The issue is not with the length of the API_SECRET but rather with what happens when the scripts are running and someone presses a few keys. Will this keys be part of the dialog, or are they removed automatically.

Reply:
The dialog will take all the keys as intended characters until the user presses enter.  Then, all the characters enterd will be accepted as the password.

Reply:
This is where that loop is right now before this PR:
![Screenshot 2023-03-18 204532](https://user-images.githubusercontent.com/51497406/226147755-50f62275-90e1-45fc-a16e-e0940534d7ca.png)


It's after the prompt asking for the password is shown in screen not before.

Reply:
This is the PR that put that loop there:  https://github.com/jamorham/nightscout-vps/pull/24

So, it's my fault.  I put it in the wrong place.
The loops should be there when a script is running taking a very long time.  Then, if the user presses a button by mistake, we need to make sure what comes next will not be broken by those keys.

But, where I have inserted the loop is wrong.